src/client/Microsoft.Identity.Client/ManagedIdentity/CredentialManagedIdentitySource.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.Http;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Internal;
+
+namespace Microsoft.Identity.Client.ManagedIdentity
+{
+    internal class CredentialManagedIdentitySource : AbstractManagedIdentity
```

Comment:
nit: maybe we just call it SLC or MSIv2 ? Credential is such an overloaded term.